powerdns/views.py

```diff
@@ -64,12 +64,19 @@ class DomainViewSet(OwnerViewSet):
     filter_fields = ('name', 'type')
     permission_classes = (DomainPermission,)
 
+    def get_queryset(self):
+        qs = super().get_queryset()
+        domain_name = self.request.query_params.get('domain_name')
```

Comment:
you have `name` in `filter_fields` - it should work (using `name` query param, not `domain_name`).